tx-pool/src/process.rs

```diff
@@ -1380,81 +1379,77 @@ fn _update_tx_pool_for_reorg(
-    tx_pool.remove_expired(detached_proposal_id.iter());
+    tx_pool.remove_committed_txs(attached.iter(), callbacks, detached_headers);
+    tx_pool.remove_by_detached_proposal(detached_proposal_id.iter());
 
     let mut entries = Vec::new();
     let mut gaps = Vec::new();
 
+    // mine mode:
     // pending ---> gap ----> proposed
     // try move gap to proposed
+    if mine_mode {
+        tx_pool.gap.remove_entries_by_filter(|id, tx_entry| {
+            if snapshot.proposals().contains_proposed(id) {
+                entries.push((
+                    Some(CacheEntry::completed(tx_entry.cycles, tx_entry.fee)),
+                    tx_entry.clone(),
+                ));
+                true
+            } else {
+                false
+            }
+        });
 
-    tx_pool.gap.remove_entries_by_filter(|id, tx_entry| {
-        if snapshot.proposals().contains_proposed(id) {
-            entries.push((
-                Some(CacheEntry::completed(tx_entry.cycles, tx_entry.fee)),
-                tx_entry.clone(),
-            ));
-            true
-        } else {
-            false
-        }
-    });
-
-    tx_pool.pending.remove_entries_by_filter(|id, tx_entry| {
-        if snapshot.proposals().contains_proposed(id) {
-            entries.push((
-                Some(CacheEntry::completed(tx_entry.cycles, tx_entry.fee)),
-                tx_entry.clone(),
-            ));
-            true
-        } else if snapshot.proposals().contains_gap(id) {
-            gaps.push((
-                Some(CacheEntry::completed(tx_entry.cycles, tx_entry.fee)),
-                tx_entry.clone(),
-            ));
-            true
-        } else {
-            false
-        }
-    });
+        tx_pool.pending.remove_entries_by_filter(|id, tx_entry| {
+            if snapshot.proposals().contains_proposed(id) {
+                entries.push((
+                    Some(CacheEntry::completed(tx_entry.cycles, tx_entry.fee)),
+                    tx_entry.clone(),
+                ));
+                true
+            } else if snapshot.proposals().contains_gap(id) {
+                gaps.push((
+                    Some(CacheEntry::completed(tx_entry.cycles, tx_entry.fee)),
+                    tx_entry.clone(),
+                ));
+                true
+            } else {
+                false
+            }
+        });
 
-    for (cycles, entry) in entries {
-        let tx_hash = entry.transaction().hash();
-        if let Err(e) = tx_pool.proposed_rtx(cycles, entry.size, entry.rtx.clone()) {
-            debug!("Failed to add proposed tx {}, reason: {}", tx_hash, e);
-            callbacks.call_reject(tx_pool, &entry, e.clone());
-        } else {
-            callbacks.call_proposed(tx_pool, &entry, false);
+        for (cycles, entry) in entries {
+            let tx_hash = entry.transaction().hash();
+            if let Err(e) = tx_pool.proposed_rtx(cycles, entry.size, entry.rtx.clone()) {
```

Comment:
the parameter `cache_entry: Option<CacheEntry>` of `proposed_rtx`  and `gap_rtx` fn may be changed to `cache_entry: CacheEntry`